File: Daily/2023/228_summaryRanges.cs
public class Solution {
    public IList<string> SummaryRanges(int[] nums) {
        int n = nums.Length;
        if (n == 0) {
            return new List<string>();
        }

        int prev = nums[0];

        List<string> ret = new List<string>();
        for (int idx = 1; idx < n + 1; idx++) {
            if (idx == n) {
                ret.Add(
                    prev == nums[idx - 1] ?
                    $"{prev}" :
                    $"{prev}->{nums[idx - 1]}"
                );
                break;
            }
            if (nums[idx] != nums[idx - 1] + 1) {
                ret.Add(
                    prev == nums[idx - 1] ?
                    $"{prev}" :
                    $"{prev}->{nums[idx - 1]}"
                );
                prev = nums[idx];
            }
        }

        return ret;
    }
}

File: Categories/Array/228_summaryRanges.py
from typing import *


class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        n = len(nums)
        if n == 0:
            return []
        prev = nums[0]

        ret = []
        for idx in range(1, n + 1):
            if idx == n:
                ret.append(
                    f"{prev}->{nums[idx - 1]}" if prev != nums[idx - 1] else
                    f"{nums[idx - 1]}"
                )
                break

            if nums[idx] != nums[idx - 1] + 1:
                ret.append(
                    f"{prev}->{nums[idx - 1]}" if prev != nums[idx - 1] else
                    f"{nums[idx - 1]}"
                )
                prev = nums[idx]
        
        return ret